packages/core/src/extension-point.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  bind,
+  BindingFromClassOptions,
+  BindingSpec,
+  BindingTemplate,
+  Constructor,
+  Context,
+  createBindingFromClass,
+  createViewGetter,
+  filterByTag,
+  inject,
+} from '@loopback/context';
+import {CoreTags} from './keys';
+
+/**
+ * Decorate a class as a named extension point. If the decoration is not
+ * present, the name of the class will be used. For example:
+ *
+ * ```ts
+ * import {extensionPoint} from '@loopback/core';
+ *
+ * @extensionPoint(GREETER_EXTENSION_POINT_NAME)
+ * export class GreetingService {
+ *   // ...
+ * }
+ * ```
+ *
+ * @param name Name of the extension point
+ */
+export function extensionPoint(name: string, ...specs: BindingSpec[]) {
+  const tags = name ? {tags: {name}} : {};
+  return bind(tags, ...specs);
+}
+
+/**
+ * Shortcut to inject extensions for the given extension point. For example:
+ *
+ * ```ts
+ * import {Getter} from '@loopback/context';
+ * import {extensionPoint, extensions} from '@loopback/core';
+ *
+ * @extensionPoint(GREETER_EXTENSION_POINT_NAME)
+ * export class GreetingService {
+ *  constructor(
+ *    @extensions() // Inject extensions for the extension point
+ *    private getGreeters: Getter<Greeter[]>,
+ *    // ...
+ * ) {
+ *   // ...
+ * }
+ * ```
+ *
+ * @param extensionPointName Name of the extension point. If not supplied, we use
+ * the name from `@extensionPoint` or the class name of the extension point
+ * class.
+ */
+export function extensions(extensionPointName?: string) {
+  return inject('', {decorator: '@extensions'}, (ctx, injection, session) => {
+    // Find the key of the target binding
+    if (!session.currentBinding) return undefined;
+
+    if (!extensionPointName) {
+      extensionPointName = session.currentBinding.tagMap.name;
```

Comment:
As I commented earlier, I find it confusing to use tag called `name` for the extension point name. Let's use a more descriptive tag name please, e.g. `extensionPointName` or simply `extensionPoint`.